pkg/controller/servicebinding/sbrcontroller.go

```diff
@@ -113,27 +114,29 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
+			dataFieldsAreEqual, err := compareRuntimeObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
 			logger.Debug("Predicate evaluated for Secret/ConfigMap", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
-		specsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "spec")
+		specsAreEqual, err := compareRuntimeObjectFields(e.ObjectOld, e.ObjectNew, "spec")
 		if err != nil {
 			logger.Error(err, "error comparing object's spec fields: %s", err.Error())
 			return false
 		}
-		statusAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "status")
+		statusAreEqual, err := compareRuntimeObjectFields(e.ObjectOld, e.ObjectNew, "status")
 		if err != nil {
 			logger.Error(err, "error comparing object's status fields: %s", err.Error())
 			return false
 		}
-		shouldReconcile := !specsAreEqual || !statusAreEqual
-		logger.Debug("Resource update event received", "GVK", e.ObjectNew.GetObjectKind(), "spec changed", !specsAreEqual, "status changed", !statusAreEqual, "should reconcile", shouldReconcile)
-
+		annOld := e.MetaOld.GetAnnotations()
+		annNew := e.MetaNew.GetAnnotations()
+		annotationsAreEqual := reflect.DeepEqual(annOld, annNew)
```

Comment:
we should trigger reconcile only if relevant annotations are changed, i.e. annotations specifying bindings.

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/704/commits/9ec2a90e4799408910298274c6fca5b1a1f29d4c